setup.py

```diff
@@ -330,6 +340,7 @@ def run(self):
     'scipy >= 0.18.1',
     'six >= 1.5.0',
     'smart_open >= 1.8.1',
+    "dataclasses; python_version < '3.7'",
```

Comment:
What does this do? Deserves a code comment.

Reply:
Added comment to explain: some of the backward compatibility code in `doc2vec.py` makes use of the py3.7-and-up `dataclass` functionality. The `dataclasses` package is a backport for py3.6. 

(When either that backward-gensim-format compatibility is dropped, or py3.6 support is dropped, this won't be necessary.)